packages/babel-plugin-proposal-class-properties/test/fixtures/private-loose/nested-class-redeclared/input.js

```diff
@@ -0,0 +1,15 @@
+class Foo {
```

Comment:
Can you add a case on redeclared private names with computed key reference? The following case should throw since `#foo` is shadowed.
```js
class Foo {
  #foo = 1;

  test() {
    class Nested {
      #foo = 2;
      [this.#foo] = 3;
    }

    this.#foo;
  }
}
```
Given that it is not directly related to this PR, we can fix it in a separate one.

Reply:
Shouldn't we wait till the other PR to add the test case? It's going to be much harder to fix.

Reply:
Actually, I have a fix.

Reply:
Oh I meant we can add the new case and the fixes in a separate PR, my bad. As long as you come up with a fix, that's awesome! 😄